env/plugin-commands-env/src/replaceOrAddNodeIntoBinPaths.ts

```diff
@@ -0,0 +1,15 @@
+import path from 'path'
+
+export function replaceOrAddNodeIntoBinPaths (binPaths: string[], baseDir: string, nodePath: string): string[] {
+  baseDir = path.resolve(baseDir)
+  if (!baseDir.endsWith(path.sep)) {
+    baseDir += path.sep
+  }
+
+  const index = binPaths.findIndex(dir => dir.startsWith(baseDir))
+  if (index < 0) {
+    return [...binPaths, nodePath]
+  } else {
+    return [...binPaths.slice(0, index), nodePath, ...binPaths.slice(index + 1)]
+  }
+}
```

Comment:
I am not sure why this complex function is needed at all. The node location should be just appended to the array. Replacing is not good because we need the specified node path to have the highest priority. What if the replaced directory didn't have the highest priority? Also, even if it will be duplicated, I don't think it is an issue.

Reply:
https://github.com/pnpm/pnpm/pull/8277/commits/0d8cb5f1fe99ca566e2dcb99f4e1ed2d1b8f94d7